UserDB.cpp

```diff
@@ -65,6 +74,20 @@ void UserDB::Start()
         std::thread update_thread([this](){ this->update_task(); });
         _inotify_thread = std::move(inotify_thread);
         _update_thread = std::move(update_thread);
+
+        int ret = sd_bus_open_system(&bus);
+        if (ret < 0) {
+            Logger::Error("Failed to connect to system bus: %s", strerror(-ret));
```

Comment:
>Logger::Error("Failed to connect to system bus: %s", strerror(-ret)); [](http://example.com/codeflow?start=12&length=69)

Continuing on this error? Won't this result in multiple errors and crashes?

Reply:
Will lead to multiple errors. Returning if the sd bus is not properly initialized.

Reply:
Consider returning an error or indication that start failed otherwise caller assumes that UserDB::Start() succeeded